### PR TITLE
Tabular: Model save/load cleanup

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

Code cleanup for model save/load functionality.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-665-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-665/1/index.html

File: autogluon/utils/tabular/ml/models/abstract/abstract_model.py [hunk 1]
```diff
@@ -28,35 +28,6 @@
...
-    if isinstance(hp_value, Categorical):
-        return hp_value[0]
-    elif isinstance(hp_value, List):
-        return [z[0] for z in hp_value]
-    elif isinstance(hp_value, NestedSpace):
-        raise ValueError("Cannot extract default value from NestedSpace. Please specify fixed value instead of: %s" % str(hp_value))
-    else:
-        return hp_value.get_hp('dummy_name').default_value
-
-
```
Comment (jwmueller) [inline autogluon/utils/tabular/ml/models/abstract/abstract_model.py:59]: Consider adding comment to code explaining what `self.model` is as I suspect this is a confusing part for those making custom models.

File: autogluon/utils/tabular/ml/models/abstract/abstract_model.py [hunk 2]
```diff
@@ -325,23 +297,60 @@ def score_with_y_pred_proba(self, y, y_pred_proba, eval_metric=None, metric_need
...
+
+        Parameters
+        ----------
+        path : str
+            Path to the saved model, minus the file name.
+            This should generally be a directory path ending with a '/' character (or appropriate path separator value depending on OS).
+            The model file is typically located in path + cls.model_file_name.
+        reset_paths : bool, default True
+            Whether to reset the self.path value of the loaded model to be equal to path.
+            It is highly recommended to keep this value as True unless accessing the original self.path value is important.
```
Comment (jwmueller) [inline autogluon/utils/tabular/ml/models/abstract/abstract_model.py:339]: maybe clarify for example when might this be important?

**Review by jwmueller — APPROVED** [review]
looks good, just had minor suggestions

File: autogluon/utils/tabular/ml/models/abstract/abstract_model.py (hunk 1 above)
Comment (Innixma) [reply autogluon/utils/tabular/ml/models/abstract/abstract_model.py:59]: Planning to do a larger doc improvement for 0.1, will do in separate PR.

File: autogluon/utils/tabular/ml/models/abstract/abstract_model.py (hunk 2 above)
Comment (Innixma) [reply autogluon/utils/tabular/ml/models/abstract/abstract_model.py:339]: Can revisit on doc overhaul for 0.1

**Pull request merged**